tests/tests_set_thin_basic.yml

```diff
@@ -0,0 +1,126 @@
+---
+- name: Snapshot a set of logical volumes across different volume groups
+  hosts: all
+  vars:
+    test_disk_min_size: "1g"
+    test_disk_count: 10
+    test_storage_pools:
+      - name: test_vg1
+        disks: "{{ range(0, 6) | map('extract', unused_disks) | list }}"
+        volumes:
+          - name: thin_pool_dev
+            thin_pool_name: thin_pool
+            thin_pool_size: '3g'
+            size: "{{ volume1_size }}"
+            thin: true
+          - name: lv2
+            thin_pool_name: thin_pool
+            size: "15"
+            thin: true
+          - name: lv3
+            thin_pool_name: thin_pool
+            size: "15%"
+            thin: true
+          - name: lv4
+            thin_pool_name: thin_pool
+            size: "15%"
+            thin: true
+      - name: test_vg2
+        disks: "{{ range(6, 10) | map('extract', unused_disks) | list }}"
+        volumes:
+          - name: lv5
+            size: "30%"
+          - name: lv6
+            size: "25%"
+          - name: lv7
+            size: "10%"
+          - name: lv8
+            size: "10%"
+    snapshot_test_set:
+      name: snapset1
+      volumes:
+        - name: snapshot VG1 LV1
+          vg: test_vg1
+          lv: lv2
+          percent_space_required: 20
+        - name: snapshot VG2 LV3
+          vg: test_vg2
+          lv: lv3
+          percent_space_required: 15
+        - name: snapshot VG2 LV4
+          vg: test_vg2
+          lv: lv4
+          percent_space_required: 15
+        - name: snapshot VG3 LV7
+          vg: test_vg3
+          lv: lv7
+          percent_space_required: 15
+  tasks:
+    - name: Run tests
+      block:
+        - name: Setup
+          include_tasks: tasks/setup.yml
+
+        - name: Run the snapshot role to create snapshot set of LVs
+          include_role:
+            name: linux-system-roles.snapshot
+          vars:
+            snapshot_lvm_action: snapshot
+            snapshot_lvm_set: "{{ snapshot_test_set }}"
+
+        - name: Assert changes for create snapset
+          assert:
+            that: snapshot_cmd["changed"]
+
+        - name: Run the snapshot role to verify the set of snapshots for the LVs
+          include_role:
+            name: linux-system-roles.snapshot
+          vars:
+            snapshot_lvm_action: check
+            snapshot_lvm_set: "{{ snapshot_test_set }}"
+            snapshot_lvm_verify_only: true
+
+        - name: Create snapset again for idempotence
+          include_role:
+            name: linux-system-roles.snapshot
+          vars:
+            snapshot_lvm_action: snapshot
+            snapshot_lvm_set: "{{ snapshot_test_set }}"
+
+        - name: Assert no changes for create snapset
+          assert:
+            that: not snapshot_cmd["changed"]
```

Comment:
This check won't work. It's impossible to check if include_role resulted in a change in Ansible. There is no proper way to check idempotency other than looking at the playbook output.

Reply:
@richm added a boolean response from the module to indicate if any action was taken.  Rich, please let me know what changes I should make.

Reply:
https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_general.html#creating-a-module
The module should return a `dict`.  That `dict` should have a field called `changed` which is `true` if the module changed something or `false` otherwise.

Reply:
> This check won't work. It's impossible to check if include_role resulted in a change in Ansible. There is no proper way to check idempotency other than looking at the playbook output.

But in this case, when the role calls this module, it uses `register: snapshot_cmd`, so the test playbook can check